packages/manager/modules/iam/src/components/actionSelect/actionSelect.controller.js

```diff
@@ -367,19 +367,56 @@ export default class ActionSelectController {
    * @param {Category} category
    * @returns {string}
    */
-  getCategoryLabel({ actions }) {
+  getCategoryLabel({ actions }, searchQuery) {
     const selectionCount = this.constructor.countSelectedActions(actions);
+    const searchCount = this.constructor.countActionsMatchingSearch(
+      actions,
+      searchQuery,
+    );
     const prefix = 'iam_action_select_category_count';
+    if (selectionCount && searchQuery) {
+      if (searchCount <= 1) {
+        return selectionCount === 1
+          ? `${prefix}_selection_one_result_one`
+          : `${prefix}_selection_many_result_one`;
+      }
+      return selectionCount === 1
+        ? `${prefix}_selection_one_result_many`
+        : `${prefix}_selection_many_result_many`;
+    }
     if (selectionCount) {
       return selectionCount === 1
         ? `${prefix}_selection_one`
         : `${prefix}_selection_many`;
     }
+    if (searchQuery) {
+      return searchCount <= 1
+        ? `${prefix}_result_one`
+        : `${prefix}_result_many`;
+    }
     return actions.length === 1
       ? `${prefix}_no_selection_one`
       : `${prefix}_no_selection_many`;
   }
 
+  filterActions(actionTree) {
+    angular.forEach(actionTree.categories, (category) => {
```

Comment:
```suggestion
    actionTree.categories.forEach( (category) => {
```
No need to use angular functions

Reply:
Done